source/cpu/cpu_power/cpupower.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpu_power
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/node-feature-discovery/source"
+)
+
+var (
+	cpuinfo     = "/host-proc/cpuinfo"
+	sysdevcpu   = "/host-sys/devices/system/cpu/cpu"
+	cpubasefreq = "/cpufreq/base_frequency"
+)
+
+// Source implements FeatureSource.
+type Source struct{}
+
+// Name returns an identifier string for this feature source.
+func (s Source) Name() string { return "cpu_power" }
+
+type NoBasefreqSupport struct {
+	message string
+}
+
+func (b *NoBasefreqSupport) Error() string {
+	return string(b.message)
+}
+
+func getcpuBaseFrequency() int {
+	baseFreq := DiscoverBaseFreq()
+	return int(baseFreq)
+}
+
+func getcpuCount() (int, error) {
+	files, err := ioutil.ReadDir("/host-sys/devices/system/cpu")
+	if err != nil {
+		return 0, err
+	}
+
+	re := regexp.MustCompile("cpu([0-9])")
+	var i int
+	for _, f := range files {
+		if re.MatchString(f.Name()) {
+			i++
+		}
+	}
+
+	return i, nil
+}
+
+func getBaseFrequency(cpu int) (int, error) {
+	var baseFreq int
+
+	baseFile := fmt.Sprintf("%s%d%s", sysdevcpu, cpu, cpubasefreq)
+	if _, err := os.Lstat(baseFile); err != nil {
+		return baseFreq, &NoBasefreqSupport{"no base frequency support"}
+	}
```

Comment:
I think you do not need this. Just try to read the file and use `os.IsNotExist()` for catching the case where file does not exist

Reply:
+1, will change it.